### ISSUE TITLE
Misc. bugs & possible improvements in managed node groups

### ISSUE THREAD

**Issue by TBeijen** [terraform-aws-modules/terraform-aws-eks#620]
# I have issues

## I'm submitting a...

* [ x] bug report
* [ ] feature request
* [ ] support request - read the [FAQ](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/faq.md) first!
* [ ] kudos, thank you, warm fuzzy

## What is the current behavior?

Some things seem not entirely correct in the recently added managed node feature (#602).

When trying to specify `source_security_group_ids` I run into this error:

```
  on .terraform/modules/eks_cluster_apps.eks_cluster/node_groups.tf line 62, in resource "random_pet" "node_groups":
  61:
  62:       lookup(each.value, "source_security_group_ids", local.workers_group_defaults["source_security_group_id"]
  63:
    |----------------
    | each.value is object with 6 attributes

Invalid value for "list" parameter: element 0: string required.
```

So, I'm doing something wrong or a it's bug. In case of the former an addition to the example would be useful.

Furthermore, only allowing a single `instance_type` seems needlessly limiting as the underlying terraform resource [accepts a list of instance types](https://www.terraform.io/docs/providers/aws/r/eks_node_group.html#instance_types).

Also, [this line](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/node_groups.tf#L93) looks like a copy-paste mistake (should lookup `source_security_group_ids` instead of `key_name`).

## If this is a bug, how to reproduce? Please include a code sample if relevant.
---

## What's the expected behavior?

No errors, clear and complete examples

## Are you able to fix this problem and submit a PR? Link here if you have already.

I'll start a WIP pr shortly.

## Environment details

* Affected module version:
* OS:
* Terraform version:

**Comment by max-rocket-internet** [comment]
Could you create an separate issue for each topic and make the title accurate? thanks 💖

**Comment by TBeijen** [comment]
Yes. Will do. Hopefully later this week I have some time to focus on this.
Consider this a placeholder.

**Comment by TBeijen** [comment]
Closing this one:

* The error was a mistake by me so N/A
* The `key_name`/`source_security_group_ids` mixup is fixed in #625
* Created #624 for overly resrtrictive instance type

Will create an issue for missing outputs. (#641 only outputs iam role arns, TF module outputs several other data)
